dotnet/samples/README.md

```diff
@@ -1 +1,121 @@
-placeholder
\ No newline at end of file
+# Samples
+
+In this folder you will find various examples showcasing the different capabilities of the .NET Teams AI Library. Here are the general instructions for setting up a sample: 
+
+## Prerequisites
+
+- [.NET 6.0 SDK](https://dotnet.microsoft.com/download/dotnet/6.0)
+- [Azure OpenAI](https://aka.ms/oai/access) resource or an account with [OpenAI](https://platform.openai.com).
+
+### Consuming the latest version of the library (preview only) 
+
+> **NOTE:**  As the library has not been published to NuGet's public registry, please complete the following steps to be able to use the samples. Otherwise, the samples will not work.
+
+The development versions of the library might not be available on NuGet's public registry. Follow the instructions below to generate the `.nukpg` NuGet package file of the library to locally consume within the sample:
+
+1. Clone the repository:
+
+`git clone https://github.com/Microsoft/teams-ai.git`
+
+2. Checkout the `DOTNET` branch.
+
+3. Navigate to the `teams-ai/dotnet/packages/Microsoft.TeamsAI` folder.
+
+4. Run `dotnet pack` in terminal.
+
+5. You should then see the following output: 
+
+`Successfully created package "C:...\teams-ai\dotnet\packages\Microsoft.TeamsAI\Microsoft.TeamsAI\bin\Debug\Microsoft.TeamsAI.1.0.0.nupkg"`
+
+6. Move the `Microsoft.TeamsAI.1.0.0.nupkg` to the `LocalPkg/` folder within the sample folder.
+
+7. Now you can install the `Microsoft.TeamsAI` package from the Visual Studio's `Manage NuGet Packages` flow or by running `dotnet add Microsoft.TeamsAI`.
+
+Now can you proceed to setting up the sample.
+
+## Setting up a sample
+
+1. Clone the repository:
+
+`git clone https://github.com/Microsoft/teams-ai.git`
+
+2. Navigate to the `teams-ai/dotnet/samples` folder, pick a sample (ex. `01.a.echoBot`) and open the `.sln` file. 
+
+By this point you should have your sample open in your IDE of choice.
+
+There are a few ways to get the application up and running. The latest way is using Teams toolkit with Visual Studio. However you can also set it up manually. You can find instructions for both below:
+
+### Using Teams Toolkit for Visual Studio
```

Comment:
I feel like we should have another third heading near the bottom for people who don't want to use TTK. There's some formatting issues that this discussion reveals: `### Manually upload the app` is a part of the TTK apps, but technically if you're doing a simple local bot, those directions are the same. 

I also think we should separate the sections using `details` -- this is to avoid customer opening a huge wall of text. We can make the TTK section open by default, and local bot directions closed by default.
![image](https://github.com/microsoft/teams-ai/assets/14900841/800c4da7-c360-437f-9a49-572b407325aa)

If you want to merge this PR in asap, I'm fine with having further discussion about this comment and filing an issue. Either way `js` system will have to be updated to match this system.
If merging asap, could you file an issue related to this comment?

Reply:
incorporated changes